DOCSP-26065: socketTimeoutMS behavior (#643)
mongodb/docs-node: source/faq.txt

milliseconds, setting the ``connectTimeoutMS`` to anything lower will
prevent the driver from ever connecting to that member.

What Happens to My Operations if My Client Disconnects?
-------------------------------------------------------

Starting in {+mdb-server+} version 4.2, the server terminates
:manual:`certain operations </release-notes/4.2/#client-disconnection>` such as
aggregations and find operations if the client disconnects before it
successfully executes an operation.

> Some operations, including all write operations, continue to run on the

Reviewer: Suggestion:

I think a transition signal word would help the reader immediately understand that this paragraph introduces the exception cases.

```suggestion
Other operations, such as write operations, continue to run on the
```

> Some operations, including all write operations, continue to run on the

Reviewer: Suggestion:

I think a transition signal word would help the reader immediately understand that this paragraph introduces the exception cases.
I think the adjective "some" doesn't seem to add any information and could be omitted.

```suggestion
Other operations, such as write operations, continue to run on the
```

{+mdb-server+} even if the client disconnects. This behavior can cause data

> inconsistencies if your application retries the operation after failure.

Reviewer: Suggestion:

I think, but am not sure, that this refers to the disconnect as the failure and not some other type of write failure. If so, it would be helpful to describe it as a disconnect since that may not be similar to a general failure.

```suggestion
inconsistencies if your application retries the operation after a disconnection.
```

How Can I Confirm That the Driver Closed Unusable Sockets?
----------------------------------------------------------

> If you experience unexpected network behavior or receive errors when
> executing a MongoDB process, you may not receive confirmation that the

Reviewer: ```suggestion
If you experience unexpected network behavior or if a MongoDB process
exits with an error, you may not receive confirmation that the
```

> driver closed the corresponding socket correctly.

Reviewer: Suggestion:
The adverb "correctly" modifies the closure, so it's usually easier to associate when placed next to word it's related to.

```suggestion
driver correctly closed the corresponding socket.
```

To make sure that the driver closes the socket correctly in these cases,

> set the ``socketTimeoutMS`` option on your ``MongoClient`` instance. Once any

Reviewer: Suggestion:

Just in case it can be on the connection string, maybe it would be more flexible to call socketTimoutMS a connection option.

```suggestion
set the ``socketTimeoutMS`` connection option. When a
```

> process times out, the driver will successfully close the socket. We

Reviewer: Suggestion:

I think it could be helpful to describe the process as a "MongoDB" one as done earlier since it's a general term.

I think it's not a definite outcome that the driver will _successfully_ close it as something else could go wrong.

```suggestion
MongoDB process times out, the driver will close the socket. We
```

recommend that you select a value for ``socketTimeoutMS`` that is two to

> three times as long as the length of the slowest operation that your

Reviewer: ```suggestion
three times as long as the expected duration of the slowest operation that your
```

milliseconds, setting the ``connectTimeoutMS`` to anything lower will
prevent the driver from ever connecting to that member.

What Happens to Running Operations If the Client Disconnects?
-------------------------------------------------------------

Starting in {+mdb-server+} version 4.2, the server terminates
certain operations such as aggregations and find operations if the

> client disconnects the operations successfully completes. To see a full

Author: eliminated a clause to shorten

list of operations affected by this behavior, see the :manual:`Server version
4.2 release notes </release-notes/4.2/#client-disconnection>` in the
Server manual.

Other operations, such as write operations, continue to run on the

> {+mdb-server+} even if the client disconnects. This behavior can cause data

Author: Not changing because it seems like the mechanism is not that the server "allows" certain operations to continue running, but rather that it marks certain operations for termination.

One message every 3000 milliseconds is not enough to keep the sockets
active, so several of the sockets will time out after 5000 milliseconds.

> Reduce the ``maxPoolSize`` option in the connection settings to lower
> the number of connection the driver can maintain in the connection
> pool.

Reviewer: Issue:

I like the intent of providing more specificity in what the user should try to accomplish by changing the setting to fix the problem. However, there are a couple minor issues:

- I think "reduce X option" doesn't seem like the right subject-verb agreement. Since it corresponds to the "lower the number of [connections]", the verb only needs to be used once.
- Also, I think "number of connection" should be "number of connections".

Suggestion:

```suggestion
To avoid excessive socket timeouts, lower the number of connections that
the driver can maintain in the connection pool, specified by the ``maxPoolSize`` 
option.
```

To specify the optional ``maxPoolSize`` setting for your ``MongoClient``, declare
it in the ``options`` object of the constructor as follows:

.. code-block:: javascript

   const client = new MongoClient(uri, {

>      maxPoolSize: <integer value>,

Reviewer: Question: does this change in the setting name apply to all versions of the driver that we document? If it applies to a different set of versions than the other changes, perhaps a different PR would work better. Alternatively, maybe setting a source constant for this with a value that differs based on version might work too.

Author: Yes, since Node v3.5 this is the correct name for this option

other, faster operations.

.. note::

>     If the number of operations is greater than the set ``maxPoolSize`` and

Reviewer: ```suggestion
    If the number of operations is greater than the ``maxPoolSize`` and
```
